arch/arm/core/CMakeLists.txt

```diff
@@ -33,3 +33,5 @@ add_subdirectory_ifdef(CONFIG_ARM_MPU cortex_m/mpu)
 add_subdirectory_ifdef(CONFIG_CPU_CORTEX_M_HAS_CMSE cortex_m/cmse)
 add_subdirectory_ifdef(CONFIG_ARM_SECURE_FIRMWARE cortex_m/tz)
 add_subdirectory_ifdef(CONFIG_ARM_NONSECURE_FIRMWARE cortex_m/tz)
+
+add_subdirectory_ifdef(CONFIG_CPU_CORTEX_R cortex_r)
```

Comment:
There seems to be an issue here, at least, with ARM_MPU, which brings-in cortex-m/mpu, but the ARM_MPU is also implied to be used in CORTEX_R (see config CPU_CORTEX_R)